src/msf/mod.rs

```diff
@@ -108,7 +120,7 @@ impl<'s, S: Source<'s>> BigMSF<'s, S> {
         // yes, this is a stupid level of indirection
         let mut stream_table_page_list_page_list = PageList::new(header_object.page_size);
         for _ in 0..size_of_stream_table_page_list_in_pages {
-            let n = header.parse_u32()?;
+            let n = bytes.gread_with(offset, LE)?;
```

Comment:
Hmm… `bytes.gread_with(offset, LE)` gets duplicated here a lot. More generally, it seems like it'll be a common pattern, particularly for replacing `ParseBuffer` code. Can we encapsulate this somehow without losing clarity?

Reply:
I thought about just rewriting `ParseBuffer` to use `gread_with` internally, which I guess would result in less code churn overall. We could give it a generic method like `fn parse<T: TryFromCtx<...>>(&mut self) -> T`

Reply:
_ponders_

Yeah, let's do that. `ParseBuffer` really is just a `(&[u8], usize)` tuple – bytes and offset, [exactly what's expanded here](https://github.com/willglynn/pdb/pull/18/files#diff-e7ae42309c9012269e74ca6f9feb8c26L78). There are a couple cases where `pread` might be more direct, but the vast majority of PDB work fits the `gread` pattern exposed by `ParseBuffer`.

Adding generic `fn parse()` to `ParseBuffer` would let us `scroll`-ify data structure reads without losing that encapsulation. I like it.

Reply:
Conveniently I started doing that shortly after writing that comment, and I have the patches ready to go. I'll push them in a minute.